app/assets/javascripts/components/discussion_index.es6.jsx

```diff
@@ -0,0 +1,153 @@
+/**
+ * @prop discussions - discussion index
+ * @prop discussion - discussion
+ * @prop current_user - current user
+ * @prop favorite_discussions - favorite user discussions
+ * @prop show_favorites - display favorites flag
+ * @prop date_handler - handler to render timestamp
+ * @prop tag_filter - tag filter param
+ */
+
+class DiscussionIndex extends React.Component {
+  
+  constructor(props) {
+    super(props);
+    this._showFavorites = this._showFavorites.bind(this);
+    this.state = {
+      show_favorites: this.props.show_favorites,
+      current_user: this.props.current_user,
+      favorite_discussions: this.props.favorite_discussions,
+      discussion: this.props.discussion,
+      discussions: this.props.discussions,
+      search: ""
+    };
+  }
+
+  _showFavorites(e) {
+    $(e.target).toggleClass('selected');
+    this.setState({ show_favorites: !this.state.show_favorites });
+  }
+
+  renderFilters() {
+    let filters = ["Starting up", "Funding", "Volunteering", "Partnering", "Learn More"];
+    return filters.map((filter, i) => {
+      var tagClass;
+      if (filter == this.props.tag_filter) {
+        tagClass = "discussion-tag checked";
+      } else {
+        tagClass = "discussion-tag";
+      }
+      return (
+        <a href={"/discussions?filter=" + filter} key={i}>
+          <button className={tagClass}>{filter}</button>
+        </a>
+      )});
+  }
+
+  renderDiscussionHeader() {
+    let searchParam = "";
+    if (this.state.search != "") {
+      searchParam = "&search=" + this.state.search;
+    }
+
+    let favParam = "";
+    if (this.state.show_favorites) {
+      favParam = "&fav=true";
+    }
+
+    let discussionParam = "";
+    if (this.state.discussion) {
+      discussionParam = "discussion_id=" + this.state.discussion.id;
+    }
+    let discussionRoute = '/discussions?' + discussionParam + searchParam + favParam;
+    let header = null;
+
+    let favoritesSelected = "";
+    if (this.state.show_favorites) {  
+      favoritesSelected = "discussion-favorite selected";
+    } else {
+      favoritesSelected = "discussion-favorite";
+    }
+    
+    if (this.state.current_user) {
+      header = (
+      <div>
```

Comment:
nit: indent this div and fix spacing (1st button element should only be indented 2 spaces)